core/controllers/topics_and_skills_dashboard.py

```diff
@@ -169,3 +179,34 @@ def post(self):
         self.render_json({
             'skillId': new_skill_id
         })
+
+
+class MergeSkillHandler(base.BaseHandler):
+    """Handles merging of the skills."""
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_access_topics_and_skills_dashboard
+    def post(self):
+        """Handles the POST request."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+        old_skill = self.payload.get('old_skill')
+        new_skill_id = self.payload.get('new_skill_id')
+        question_services.update_skill_ids_of_questions(
+            old_skill['id'], new_skill_id)
+        changelist = [
+            skill_domain.SkillChange({
+                'cmd': skill_domain.CMD_UPDATE_SKILL_PROPERTY,
+                'property_name': (
+                    skill_domain.SKILL_PROPERTY_ALL_QUESTIONS_MERGED),
+                'old_value': False,
+                'new_value': True
+            })
+        ]
+        skill_services.update_skill(
+            self.user_id, old_skill['id'], changelist,
+            'Setting merge complete for skill.')
```

Comment:
Perhaps: 'Marking the skill as having being merged successfully.'

Reply:
@vinitamurthi did you see this comment?

Reply:
Made the change